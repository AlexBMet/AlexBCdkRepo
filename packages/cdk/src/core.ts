export const generateResourcePrefix = (uniquePrefix: string, environment: string, region: string) => {
	return `${uniquePrefix}-${environment}-${region}`
};
