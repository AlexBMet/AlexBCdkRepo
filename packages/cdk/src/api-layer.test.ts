import {SynthUtils} from '@aws-cdk/assert'; // tslint:disable-line:no-implicit-dependencies
import {Stack} from '@aws-cdk/core';
import {APILayer} from './api-layer';

describe('[api-layer.ts] unit tests', () => {

	describe('[stack]', () => {
		it('must create an lambda function', async () => {
			const stack = new APILayer(new Stack(), 'APILayerStack', {});
			expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
		});
	});
});
