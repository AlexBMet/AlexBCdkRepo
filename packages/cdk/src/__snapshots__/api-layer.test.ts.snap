// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[api-layer.ts] unit tests [stack] must create an lambda function 1`] = `
Object {
  "Parameters": Object {
    "Environment": Object {
      "AllowedValues": Array [
        "dev",
        "ci",
        "stg",
        "prod",
      ],
      "Description": "TBC",
      "Type": "String",
    },
    "ServiceCode": Object {
      "Description": "TBC",
      "Type": "String",
    },
    "ServiceName": Object {
      "Description": "TBC",
      "Type": "String",
    },
    "ServiceOwner": Object {
      "Description": "TBC",
      "Type": "String",
    },
    "SourceBucketName": Object {
      "Description": "TBC",
      "Type": "String",
    },
    "SourceObjectKey": Object {
      "Description": "TBC",
      "Type": "String",
    },
    "UniquePrefix": Object {
      "Description": "TBC",
      "Type": "String",
    },
  },
  "Resources": Object {
    "LambdaFunctionBF21E41F": Object {
      "DependsOn": Array [
        "LambdaFunctionServiceRoleDefaultPolicy32EEEE35",
        "LambdaFunctionServiceRoleC555A460",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "SourceBucketName",
          },
          "S3Key": Object {
            "Ref": "SourceObjectKey",
          },
        },
        "Description": "TBC",
        "FunctionName": Object {
          "Fn::Join": Array [
            "",
            Array [
              Object {
                "Ref": "UniquePrefix",
              },
              "-",
              Object {
                "Ref": "Environment",
              },
              "-",
              Object {
                "Ref": "AWS::Region",
              },
              "-lambda",
            ],
          ],
        },
        "Handler": "dist/index.handler",
        "MemorySize": 128,
        "Role": Object {
          "Fn::GetAtt": Array [
            "LambdaFunctionServiceRoleC555A460",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
        "Tags": Array [
          Object {
            "Key": "Environment",
            "Value": Object {
              "Ref": "Environment",
            },
          },
          Object {
            "Key": "ServiceCode",
            "Value": Object {
              "Ref": "ServiceCode",
            },
          },
          Object {
            "Key": "ServiceName",
            "Value": Object {
              "Ref": "ServiceName",
            },
          },
          Object {
            "Key": "ServiceOwner",
            "Value": Object {
              "Ref": "ServiceOwner",
            },
          },
        ],
        "Timeout": 30,
        "TracingConfig": Object {
          "Mode": "Active",
        },
      },
      "Type": "AWS::Lambda::Function",
    },
    "LambdaFunctionServiceRoleC555A460": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
        "Tags": Array [
          Object {
            "Key": "Environment",
            "Value": Object {
              "Ref": "Environment",
            },
          },
          Object {
            "Key": "ServiceCode",
            "Value": Object {
              "Ref": "ServiceCode",
            },
          },
          Object {
            "Key": "ServiceName",
            "Value": Object {
              "Ref": "ServiceName",
            },
          },
          Object {
            "Key": "ServiceOwner",
            "Value": Object {
              "Ref": "ServiceOwner",
            },
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "LambdaFunctionServiceRoleDefaultPolicy32EEEE35": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "xray:PutTraceSegments",
                "xray:PutTelemetryRecords",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "LambdaFunctionServiceRoleDefaultPolicy32EEEE35",
        "Roles": Array [
          Object {
            "Ref": "LambdaFunctionServiceRoleC555A460",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "LambdaLogs02F6D053": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "LogGroupName": Object {
          "Fn::Join": Array [
            "",
            Array [
              "/aws/lambda/",
              Object {
                "Ref": "LambdaFunctionBF21E41F",
              },
            ],
          ],
        },
        "RetentionInDays": 3,
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Delete",
    },
  },
}
`;
